File: frontend/src/api/stats.ts
import { apiRequest } from './client'

export type MonthlyStats = {
  month: string
  total_spend: string
  total_income: string
  breakdown: Array<{ category_id: number; category_name: string; type: string; total: string }>
  budget_vs_actual: Array<{
    category_id: number
    category_name: string
    limit_amount: string
    actual_amount: string
  }>
}

export async function getMonthlyStats(month: string) {
  const res = await apiRequest<MonthlyStats>(`/stats/monthly?month=${month}`)
  return res.data
}

File: backend/app/schemas/stats.py
from decimal import Decimal

from pydantic import BaseModel


class CategoryBreakdown(BaseModel):
    category_id: int
    category_name: str
    type: str
    total: Decimal


class BudgetActualItem(BaseModel):
    category_id: int
    category_name: str
    limit_amount: Decimal
    actual_amount: Decimal


class MonthlyStatsResponse(BaseModel):
    month: str
    total_spend: Decimal
    total_income: Decimal
    breakdown: list[CategoryBreakdown]
    budget_vs_actual: list[BudgetActualItem]
